Flaky unit test `TestEvents/multiple_emitters`

Fails on CI from time to time:

```
 --- FAIL: TestEvents (1.04s)
    --- FAIL: TestEvents/multiple_emitters (0.01s)
        events_test.go:144: 
            	Error Trace:	/go/src/github.com/k0sproject/k0s/pkg/component/prober/events_test.go:144
            	Error:      	"[{2023-03-02 10:57:11.211543598 +0000 UTC m=+1.029813166 message1 <nil>} {2023-03-02 10:57:11.211544686 +0000 UTC m=+1.029814252 message2 <nil>}]" should have 3 item(s), but has 2
            	Test:       	TestEvents/multiple_emitters
FAIL
FAIL	github.com/k0sproject/k0s/pkg/component/prober	1.072s
```

--- a/pkg/component/prober/prober.go
+++ b/pkg/component/prober/prober.go
@@ -81,6 +81,7 @@ func (p *Prober) State(maxCount int) State {
 		Events:       make(map[string][]Event),
 	}
 	for name, r := range p.healthCheckState {
+		maxCount := maxCount
 		state.HealthProbes[name] = make([]ProbeResult, 0, p.probesTrackLength*len(p.withHealthComponents))
 		r.Do(func(v interface{}) {
 			if v == nil {
@@ -97,6 +98,7 @@ func (p *Prober) State(maxCount int) State {
 		state.HealthProbes[name] = state.HealthProbes[name][0:maxCount]
 	}
 	for name, r := range p.eventState {
+		maxCount := maxCount
 		state.Events[name] = make([]Event, 0, p.eventsTrackLength*len(p.withEventComponents))
 		r.Do(func(v interface{}) {
 			if v == nil {

